account/account.go

```diff
@@ -40,7 +40,6 @@ type AccountInterface interface {
 }
 
 var _ AccountInterface = &Account{}
-var _ rpc.RpcProvider = &Account{}
```

Comment:
Imo we should keep this in. If a new method is added to the RPC spec then this helps ensure that the account type implements it. What do you think @thiagodeev

Reply:
Otherwise the PR looks good

Reply:
This line was removed because as the AddDeclare/Invoke/DeployAcc methods were removed from the Account type it no longer implements the RPC interface.
IMO, the Account type doesn't need to implement the RPC interface because they are separate abstractions. If the user wants to use an RPC method it can just access the 'provider' Account type property. I see this approach in the Ethers.js library https://docs.ethers.org/v6/api/providers/#Signer. 
We can adapt some provider methods to the context of the Account (e.g. Nonce could return the account nonce instead of being a duplicated version of the provider.Nonce) but no need to have all of them IMO.